jena-arq/src/main/java/org/apache/jena/riot/RDFWriterRegistry.java

```diff
@@ -82,24 +82,21 @@ public WriterGraphRIOT create(RDFFormat serialization)
     }} ;
 
     // Writing a dataset
-    static WriterDatasetRIOTFactory wdsfactory = new WriterDatasetRIOTFactory() {
-        @Override
-        public WriterDatasetRIOT create(RDFFormat serialization)
-        {
-            if ( Objects.equals(RDFFormat.TRIG_PRETTY, serialization) )
-                return new TriGWriter() ;
-            if ( Objects.equals(RDFFormat.TRIG_BLOCKS, serialization) )
-                return new TriGWriterBlocks() ;
-            if ( Objects.equals(RDFFormat.TRIG_FLAT, serialization) )
-                return new TriGWriterFlat() ;
-            if ( Objects.equals(RDFFormat.NQUADS_UTF8, serialization) )
-                return new NQuadsWriter() ;
-            if ( Objects.equals(RDFFormat.NQUADS_ASCII, serialization) )
-                return new NQuadsWriter(CharSpace.ASCII) ;
-            if ( Objects.equals(RDFFormat.RDFNULL, serialization) )
-                return NullWriter.factory.create(RDFFormat.RDFNULL) ;
-            return null ;
-    }} ;
+    static WriterDatasetRIOTFactory wdsfactory = (RDFFormat serialization) -> {
+        if ( Objects.equals(RDFFormat.TRIG_PRETTY, serialization) )
+            return new TriGWriter() ;
+        if ( Objects.equals(RDFFormat.TRIG_BLOCKS, serialization) )
+            return new TriGWriterBlocks() ;
+        if ( Objects.equals(RDFFormat.TRIG_FLAT, serialization) )
```

Comment:
Would it be convenient for the future to let `RDFFormat` feature methods to get writers and readers? Then `TRIG_BLOCKS` has `writer() { return new TriGWriterBlocks(); }` etc. Happy to send an independent PR if so.

Reply:
These few lines are only in the PR as a small bit of clearup to switch to using a function not an anonymous class.

What capability does that give that isn't available currently?

The right way to get a writer is via `RDFWriter.create()`. 

`RDFFormat` doesn't have readers.

One writer can service several `RDFFormats` - as happens for current JSON-LD.

The registry+factory approach means the association can be altered dynamically. We can use this to - for example - make `Lang.JSONLD`/`RDFFormat.JSONLD` switch from jsonld-java (1.0) and titanium (1.1).

Reply:
I can see your point about closed vs open-- I guess I haven't ever added a format or had any reason to. In any event, this is a tiny thing and I am happy to leave it with your deep knowledge of the (de)ser systems. I haven't ever used them enough to understand where the proper entry points are.

Reply:
These formats (and more generally `Lang`) are a way to have "type checked constants". The use of string for everything causes problems e.g. this week a question about `model.read(String, String)` where the second string is base URI, not syntax. A legacy thing - we are where we are.

`Lang` in particular,may not even be implemented. `Lang` also cover result sets.

It's a bit clunky in Java but there again, a programming language with all possible features makes long-term readability suffer.